igvm/hypervisor.py

```diff
@@ -1194,9 +1197,32 @@ def log_migration(self, vm: VM, operator: str) -> None:
         :param operator: plus for migration to HV, minus for migration from HV
         """
 
-        cpu_usage_vm = self.estimate_vm_cpu_usage(vm)
-        timestamp = int(time())
-        log_entry = '{} {}{}'.format(timestamp, operator, round(cpu_usage_vm))
-
-        self.dataset_obj['igvm_migration_log'].add(log_entry)
-        self.dataset_obj.commit()
+        retry = 5
+        for i in range(0, retry):
+            # It can happen, that multiple migrations at the same time will write the same log
+            # entry. This will fail during the commit and throw a DatasetError
+            try:
+                cpu_usage_vm = self.estimate_vm_cpu_usage(vm)
+                timestamp = int(time())
+                log_entry = '{} {}{}'.format(timestamp, operator, round(cpu_usage_vm))
+                self.dataset_obj['igvm_migration_log'].add(log_entry)
+                self.dataset_obj.commit()
+                return
+            except DatasetError:
+                log.error(
+                    'Failed to log migration for VM "{}" on Hypervisor "{}". '
+                    'Will try again...'.format(
+                        str(vm),
+                        str(self),
+                    )
+                )
+                sleep(1)
```

Comment:
Just use the number of retry (i) as sleep duration this way the back of time in between increases which makes it a bit more graceful.